Smartcar and SMCE references

bassamEldesouki opened the pull request:
"Badge is added for Arduino CI integration"
- Project logo is added
- Rearranged the readMe.md file by adding the followings;
  - Index shortcuts
  - Introduction modified
  - Dependency links are added
  - More detailed Installation section by adding wiki pages for the user, developer guidance for macOS & Windows.
  - Developers and their links added.
  - Testing links added to readMe.md and wiki pages have been created.
  - Credit & License links added to readMe.md and wiki pages have been created.



Resolves #92 
Resolves #93 
Resolves #81 



Co-authored-by: @edizgenc2021 <gusgened@student.gu.se>

bassamEldesouki committed:
Smartcar and SMCE references 

Add the smartcar shield and SMCE godot references, so we declare our sources and also for the user to be able to run the application and try it on the emulator.  

Resolves #81

ediz-genc reviewed (dismissed):
Since we didn't have the right issue# when you created this PR, no objection ;) Now we have the right issue related to this PR, you may just easily edit your resolve#

iachella reviewed (commented):
Looks good! Maybe at line 2 we can mention that when cloning the repository you get both the app and the smartcar sketch

bassamEldesouki commented:
> Looks good! Maybe at line 2 we can mention that when cloning the repository you get both the app and the smartcar sketch

Yes, and maybe also post some screenshoots where to find it and how to run the sketch on the SMCE! Will fix it :)

ediz-genc committed:
Adding Arduino CI badges

Following CI badges are added;
"Push & Pull-requests" for building and testing the sketch

Resolves #51 
Resolves #92 
Closes #51 



Co-authored-by: @bassamEldesouki <bassam.eldesouki.k@gmail.com>

ediz-genc committed:
Merge remote-tracking branch 'origin/master' into Readme-file

ediz-genc committed:
"Arduino build & test .yml files combined"
"Project name is added into CMakeList.txt"

Resolves #51
Closes #51

Co-authored-by: @bassamEldesouki <bassam.eldesouki.k@gmail.com>

ediz-genc committed:
Update for ReadMe.md

"Badge is added for Arduino CI integration"
- Project logo is added
- Rearranged the readMe.md file by adding the followings;
  - Index shortcuts
  - Introduction modified
  - Dependency links are added
  - More detailed Installation section by adding wiki pages for user, developer guidance for MacOS & Windows.
  - Developers and their links added.
  - Testing links added to readMe.md and wiki pages have been created.
  - Credit & License links added to readMe.md and wiki pages have been created.

Resolves #92 
Resolves #93 
Resolves #81 



Co-authored-by: @bassamEldesouki <bassam.eldesouki.k@gmail.com>

ediz-genc committed:
Edited readme header

Logo and header aligned and modified.

Resolves #92 



Co-authored-by: @bassamEldesouki <bassam.eldesouki.k@gmail.com>

ediz-genc committed:
"Fixed typos in Readme.md"

Resolves #92

Co-authored-by: @bassamEldesouki <bassam.eldesouki.k@gmail.com>

juliavankirk commented on README.md:
```diff
@@ -1,9 +1,21 @@
 
+***
+
+## SMCE godot Installation
```
Should we also add an installation guide for Android Studio itself, since we are assuming the User first needs to download SMCE as well?

ediz-genc replied:
That has been done in Wiki page.

juliavankirk commented on README.md:
```diff
@@ -16,7 +16,7 @@
 
 ## Introduction
-The purpose of the PathFinder smart vehicle is to offer to public transportation companies and their client a digitalized experience. We aim to do so by offering:
+The purpose of the PathFinder smart vehicle is to offer to public transportation companies, and their client a digitalized experience. We aim to do so by offering:
```
Minor change suggestion:

The purpose of the PathFinder smart vehicle is to offer a digitalized experience to public transportation companies and their customers.

ediz-genc replied:
Done

juliavankirk commented on README.md:
```diff
@@ -45,7 +45,7 @@ Please refer below pages  for a complete tutorial of the system;
 ## Smartcar_Shield
 
-The samrtcar functionalities in Pathfinder is using the open source library [smartcar_shield](https://github.com/platisd/smartcar_shield).
+The smart-car functionalities in Pathfinder is using the open source library [smartcar_shield](https://github.com/platisd/smartcar_shield).
```
Minor change suggestion:

The smart-car functionalities in Pathfinder utilizes the open source library [smartcar_shield](https://github.com/platisd/smartcar_shield).

ediz-genc replied:
Done

juliavankirk reviewed (approved):
Minor changes suggested, but otherwise approved

juliavankirk commented on CMakeLists.txt:
```diff
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.13)
-project(CppProject)
+project(PartFinder)
```
Should this be “Pathfinder”?

ediz-genc replied:
Good catch lol

ediz-genc replied:
Done

ediz-genc committed:
"Fixed typos in Readme.md & CMakeList.txt"
"Pull-request badge is added"
"GUI_CI.yml changed to Android_CI.yml"

Resolves #92
Resolves #93

Co-authored-by: @bassamEldesouki <bassam.eldesouki.k@gmail.com>

iachella commented on README.md:
```diff
@@ -1,10 +1,73 @@
+- [Testing](#testing)
+- [What](#what?)
+- [Why](#why?)
+- [How](#howt?)
```
this link might not work , should it be (#how?)

ediz-genc replied:
It should be fixed now.

iachella reviewed (commented):
Looks good! Just left a comment for a link which might need to be fixed. 
Good work!

ediz-genc committed:
"Links are fixed"

Resolves #92
Resolves #93

Co-authored-by: @bassamEldesouki <bassam.eldesouki.k@gmail.com>

juliavankirk commented on README.md:
```diff
@@ -30,12 +30,14 @@ to passenger's interaction with the bus directly through their phone.
 
+Installation guidance are provided in detail on the Wiki pages below.
+
 - To install the application, please proceed to [user installation](https://github.com/DIT112-V21/group-07/wiki/User-installation-&-setup-guidance).
```
Do we need User Installation vs **Developer Installation**? If someone's a developer, we should advise on how to fork the repo for them to contribute or some other strategy. Perhaps all necessary set up instructions are covered under **User Installation**, rendering the former redundant? Just a thought

ediz-genc replied:
Maybe, developer installation is on Wiki. If you have the idea, you can just edit it

alrapal committed:
Update README.md

Added a link to an android unit testing page

alrapal commented on README.md:
```diff
@@ -1,10 +1,75 @@
+
+To run the unit tests for the back-end, please refer the link below;
+ - [Arduino unit testing](https://github.com/DIT112-V21/group-07/wiki/Guidance-for-unit-testing)
+
```
I did a commit where I added a link to Android testing for consistancy.

alrapal reviewed (approved):
Good job!
It looks complete!
Just added a link to android testing as you can see below. (and in the next commit)
